homeassistant/components/media_player/_reproduce_state.py

```diff
@@ -0,0 +1,45 @@
+"""Module that groups code required to handle state restore for component."""
+from typing import Iterable
+
+from homeassistant.const import (
+    ATTR_STATE, SERVICE_MEDIA_PAUSE, SERVICE_MEDIA_PLAY, SERVICE_MEDIA_SEEK,
+    SERVICE_MEDIA_STOP, SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_VOLUME_MUTE,
+    SERVICE_VOLUME_SET, STATE_IDLE, STATE_OFF, STATE_ON, STATE_PAUSED,
+    STATE_PLAYING)
+from homeassistant.core import State
+from homeassistant.helpers.state import (
+    async_construct_service_schemas, async_reproduce_domain_state)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.loader import bind_hass
+
+
+@bind_hass
+async def async_reproduce_states(hass: HomeAssistantType,
+                                 states: Iterable[State]) -> None:
+    """Reproduce component states."""
+    from . import (
```

Comment:
Let's move these constants to `media_player/const.py`

Reply:
Ie like the following to start with. So i don't have to update all other components potentially using the constants.

```patch
diff --git a/homeassistant/components/media_player/__init__.py b/homeassistant/components/media_player/__init__.py
index dbf5d9e..716d3de 100644
--- a/homeassistant/components/media_player/__init__.py
+++ b/homeassistant/components/media_player/__init__.py
@@ -35,12 +35,12 @@ from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.entity_component import EntityComponent
 from homeassistant.loader import bind_hass
 
+from .const import *
 from .reproduce_state import async_reproduce_states  # noqa
 
 _LOGGER = logging.getLogger(__name__)
 _RND = SystemRandom()
 
-DOMAIN = 'media_player'
 DEPENDENCIES = ['http']
 
 ENTITY_ID_FORMAT = DOMAIN + '.{}'
@@ -56,38 +56,6 @@ ENTITY_IMAGE_CACHE = {
     CACHE_MAXSIZE: 16
 }
 
-SERVICE_PLAY_MEDIA = 'play_media'
-SERVICE_SELECT_SOURCE = 'select_source'
-SERVICE_SELECT_SOUND_MODE = 'select_sound_mode'
-SERVICE_CLEAR_PLAYLIST = 'clear_playlist'
-
-ATTR_MEDIA_VOLUME_LEVEL = 'volume_level'
-ATTR_MEDIA_VOLUME_MUTED = 'is_volume_muted'
-ATTR_MEDIA_SEEK_POSITION = 'seek_position'
-ATTR_MEDIA_CONTENT_ID = 'media_content_id'
-ATTR_MEDIA_CONTENT_TYPE = 'media_content_type'
-ATTR_MEDIA_DURATION = 'media_duration'
-ATTR_MEDIA_POSITION = 'media_position'
-ATTR_MEDIA_POSITION_UPDATED_AT = 'media_position_updated_at'
-ATTR_MEDIA_TITLE = 'media_title'
-ATTR_MEDIA_ARTIST = 'media_artist'
-ATTR_MEDIA_ALBUM_NAME = 'media_album_name'
-ATTR_MEDIA_ALBUM_ARTIST = 'media_album_artist'
-ATTR_MEDIA_TRACK = 'media_track'
-ATTR_MEDIA_SERIES_TITLE = 'media_series_title'
-ATTR_MEDIA_SEASON = 'media_season'
-ATTR_MEDIA_EPISODE = 'media_episode'
-ATTR_MEDIA_CHANNEL = 'media_channel'
-ATTR_MEDIA_PLAYLIST = 'media_playlist'
-ATTR_APP_ID = 'app_id'
-ATTR_APP_NAME = 'app_name'
-ATTR_INPUT_SOURCE = 'source'
-ATTR_INPUT_SOURCE_LIST = 'source_list'
-ATTR_SOUND_MODE = 'sound_mode'
-ATTR_SOUND_MODE_LIST = 'sound_mode_list'
-ATTR_MEDIA_ENQUEUE = 'enqueue'
-ATTR_MEDIA_SHUFFLE = 'shuffle'
-
 MEDIA_TYPE_MUSIC = 'music'
 MEDIA_TYPE_TVSHOW = 'tvshow'
 MEDIA_TYPE_MOVIE = 'movie'
diff --git a/homeassistant/components/media_player/reproduce_state.py b/homeassistant/components/media_player/reproduce_state.py
index 46352e3..5feafad 100644
--- a/homeassistant/components/media_player/reproduce_state.py
+++ b/homeassistant/components/media_player/reproduce_state.py
@@ -11,25 +11,26 @@ from homeassistant.core import Context, State
 from homeassistant.helpers.typing import HomeAssistantType
 from homeassistant.loader import bind_hass
 
+from .const import (
+    ATTR_MEDIA_VOLUME_LEVEL,
+    ATTR_MEDIA_VOLUME_MUTED,
+    ATTR_MEDIA_SEEK_POSITION,
+    ATTR_INPUT_SOURCE,
+    ATTR_SOUND_MODE,
+    ATTR_MEDIA_CONTENT_TYPE,
+    ATTR_MEDIA_CONTENT_ID,
+    ATTR_MEDIA_ENQUEUE,
+    SERVICE_PLAY_MEDIA,
+    SERVICE_SELECT_SOURCE,
+    SERVICE_SELECT_SOUND_MODE,
+    DOMAIN,
+)
+
 
 async def _async_reproduce_states(hass: HomeAssistantType,
                                   state: State,
                                   context: Optional[Context] = None) -> None:
     """Reproduce component states."""
-    from . import (
-        ATTR_MEDIA_VOLUME_LEVEL,
-        ATTR_MEDIA_VOLUME_MUTED,
-        ATTR_MEDIA_SEEK_POSITION,
-        ATTR_INPUT_SOURCE,
-        ATTR_SOUND_MODE,
-        ATTR_MEDIA_CONTENT_TYPE,
-        ATTR_MEDIA_CONTENT_ID,
-        ATTR_MEDIA_ENQUEUE,
-        SERVICE_PLAY_MEDIA,
-        SERVICE_SELECT_SOURCE,
-        SERVICE_SELECT_SOUND_MODE,
-        DOMAIN,
-    )
 
     async def call_service(service: str, keys=()):
         data = {}

```